pkg/ext-proc/scheduling/scheduler.go

```diff
@@ -0,0 +1,54 @@
+// Package scheduling implements request scheduling algorithms.
+package scheduling
+
+import (
+	"math/rand"
+
+	klog "k8s.io/klog/v2"
+
+	"ext-proc/backend"
+)
+
+var (
+	defaultFilter = &filter{
+		name:   "least queuing",
```

Comment:
Perhaps for later, but I feel that we will need to score more than strictly filter.

Here for example, limiting the search to the servers with the shortest queue doesn't seem optimal, what if the servers with the shortest queue don't have the adapter, and the next shortest ones do (and they are not significantly shorter)?

Reply:
The filter is flexible that instead of filtering the absolute shortest, you can select a range. That's what I did for the "least KV cache".

But agree that we may want some scoring or sorting later.

Reply:
sounds good